membership_extension/models/membership_line.py

```diff
@@ -42,13 +42,17 @@ def _onchange_membership_date(self):
                 if date_to >= self.date:
                     self.date_to = date_to
 
-    @api.depends("account_invoice_id.invoice_payment_state")
     def _compute_state(self):
         for line in self:
-            if isinstance(line.id, models.NewId):
+            if isinstance(line.id, models.NewId) or not line.account_invoice_id:
                 line.state = line.state or "none"
-            if line.account_invoice_id.invoice_payment_state == "paid":
-                line.state = "paid"
+            elif (
+                line.account_invoice_id.state == "posted"
+                and line.account_invoice_id.payment_state == "reversed"
+            ):
+                line.state = "canceled"
+            else:
+                super(MembershipLine, line)._compute_state()
```

Comment:
Looking in the core method, I wonder if this override is needed anymore: https://github.com/odoo/odoo/blob/14.0/addons/membership/models/membership.py#L48-L78

Reply:
The case of invoice refunded is not covered in core method and it fixes the possible issue when `account_invoice_id`  is null

https://github.com/odoo/odoo/blob/ff6d0b16cd32a5ed56a2cc039f5226dc56f6db6a/addons/membership/models/membership.py#L48-L63